src/node_platform.cc

```diff
@@ -462,10 +462,10 @@ void NodePlatform::DrainTasks(Isolate* isolate) {
   std::shared_ptr<PerIsolatePlatformData> per_isolate = ForNodeIsolate(isolate);
   if (!per_isolate) return;
 
-  do {
-    // Worker tasks aren't associated with an Isolate.
-    worker_thread_task_runner_->BlockingDrain();
-  } while (per_isolate->FlushForegroundTasksInternal());
+  // Drain foreground tasks but not worker tasks as this may cause deadlocks
```

Comment:
```suggestion
  // Worker tasks aren't associated with an Isolate.
  worker_thread_task_runner_->BlockingDrain();
  // Drain foreground tasks but not worker tasks as this may cause deadlocks
```